src/pymorize/units.py

```diff
@@ -9,189 +9,253 @@
+    s : str
+        The unit string to parse.
+    pattern : re.Pattern, optional
+        The regular expression pattern to use for searching for chemical
+        symbols in the unit string. Defaults to a pattern that matches
+        "mol" followed by any number of word characters.
+
+    Returns
+    -------
+    None
 
     Raises
     ------
     ValueError
-        If the chemical element is not found in the periodic table.
-
-    See Also
-    --------
-    ~chemicals.elements.periodic_table: Periodic table of elements
-    ~re.compile: `Python's regex syntax <https://docs.python.org/3/library/re.html#regular-expression-syntax>`_.
+        If the chemical symbol is not recognized.
     """
     if s is None:
         return
-    match = pattern.search(s)
-    if match:
-        d = match.groupdict()
-        try:
-            element = getattr(periodic_table, d["symbol"])
-        except AttributeError:
-            raise ValueError(
-                f"Unknown chemical element {d.groupdict()['symbol']} in {d.group()}"
-            )
-        else:
+    try:
+        ureg(s)
+    except pint.errors.UndefinedUnitError:
+        if match := pattern.search(s):
+            d = match.groupdict()
             try:
-                ureg(s)
-            except pint_xarray.pint.errors.UndefinedUnitError:
+                element = getattr(periodic_table, d["symbol"])
+            except AttributeError:
+                raise ValueError(
+                    f"Unknown chemical element {d['symbol']} in {match.group()}"
+                )
+            else:
                 logger.debug(f"Chemical element {element.name} detected in units {s}.")
                 logger.debug(
                     f"Registering definition: {match.group()} = {element.MW} * g"
                 )
                 ureg.define(f"{match.group()} = {element.MW} * g")
 
 
-def handle_unit_conversion(da: xr.DataArray, rule: Rule) -> xr.DataArray:
-    """Performs the unit-aware data conversion.
+def handle_scalar_units(da, from_unit, to):
+    """
+    Convert a DataArray with scalar units from one unit to another.
 
-    If `source_unit` is provided, it is used instead of the unit from DataArray.
+    This function handles the conversion of a `xarray.DataArray` containing
+    scalar units to another unit. The function uses the `pint` library for
+    unit conversion. If the initial quantification fails due to an undefined
+    unit, it attempts to assign and quantify the unit manually.
 
     Parameters
     ----------
-    da: ~xr.DataArray
-    unit: str
-        unit to convert data to
+    da : xarray.DataArray
+        The DataArray to be converted.
+    from_unit : str
+        The unit of the input DataArray.
+    to : str
+        The unit to convert the DataArray to.
 
     Returns
     -------
-    ~xr.DataArray
-        DataArray with units converted to `unit`.
+    xarray.DataArray
+        The converted DataArray with the new unit.
```

Comment:
Please check here, using the `~` generates a clickable cross-link in the documentation, I'd like to preserve that behaviour. Maybe it also works with the full name, I am not sure.

Reply:
fixed.